algo-coding-exercise/getting-started/src/main/java/codingblackfemales/gettingstarted/MovingAverage.java

```diff
@@ -0,0 +1,155 @@
+package codingblackfemales.gettingstarted;
+
+import codingblackfemales.action.Action;
+import codingblackfemales.action.CancelChildOrder;
+import codingblackfemales.action.CreateChildOrder;
+import codingblackfemales.action.NoAction;
+import codingblackfemales.algo.AlgoLogic;
+import codingblackfemales.sotw.ChildOrder;
+import codingblackfemales.sotw.SimpleAlgoState;
+import codingblackfemales.sotw.marketdata.AskLevel;
+import codingblackfemales.sotw.marketdata.BidLevel;
+import codingblackfemales.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import messages.order.Side;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+
+public class MovingAverage implements AlgoLogic {
+
+    private static final Logger logger = LoggerFactory.getLogger(MovingAverage.class);
+
+    int SHORT_TERM_PERIOD = 3;
+    int LONG_TERM_PERIOD = 7;
+
+    List<Long> askHistoricalPrices = new ArrayList<>();
+    long filledQuantity = 0;
+    long buyAtPrice = 0;
+
+    long counter = 0;
+
+    String side = "";
+
+    /***
+     *
+     * @param state represents the current market data and the trading environment
+     * @return an action e.g. createChildOrder, cancelChildOrder or
+     *         NoAction.NoAction etc.
+     */
+    @Override
+    public Action evaluate(SimpleAlgoState state) {
+        final String orderBookAsString = Util.orderBookToString(state);
+        logger.info("[MOVINGAVERAGE] The state of the order book is:\n" + orderBookAsString);
+        final BidLevel bidLevel = state.getBidAt(0);
+        final AskLevel askLevel = state.getAskAt(0);
+        long bidQuantity = 100;
+
+        var totalOrderCount = state.getChildOrders().size();
+
+        if (totalOrderCount > 29) {
+            logger.info("[MOVINGAVERAGE]: we have " + totalOrderCount + " orders in the market. Want " + totalOrderCount
+                    + " done.");
+            return NoAction.NoAction;
+        }
+
+        if (askLevel != null && bidLevel != null) {
+            long askLevelPrice = askLevel.price;
+
+            List<Long> gatherAskHistoricalPrices = getHistoricalPrices(askHistoricalPrices, askLevelPrice);
```

Comment:
can we change this variable name, it sounds more like a function.

Also why not assign back to the original ArrayList `askHistoricalPrices`?